webapp/src/components/NewsList/NewsList.vue

```diff
@@ -51,10 +52,23 @@ export default {
       });
       return newId;
     },
+    reverseOrder: function () {
+      this.descending = !this.descending;
+    },
   },
   computed: {
     orderedItems: function () {
-      return this.items.sort(this.compareVotes);
+      let orderedItems = [...this.items];
+      let compareVotes = (a, b) => {
+        if (a.votes < b.votes) return 1;
+        if (a.votes > b.votes) return -1;
+        return 0;
+      };
+      if (this.descending) {
+        return orderedItems.sort(compareVotes);
+      } else {
+        return orderedItems.sort(compareVotes).reverse();
```

Comment:
For fans of the ternary operator:
```suggestion
      return this.descending ? orderedItems.sort(compareVotes) : orderedItems.sort(compareVotes).reverse();
```
Better yet: Select different comparator for `compareVotes` based on `this.descending`; saves additional computation on `reverse()` :smiley:

Reply:
Or do it this way:
```js
orderedItems = orderedItems.sort(compareVotes);
if (this.descending) orderedItems = orderedItems.reverse()
return orderedItems
```